src/firebolt_db/firebolt_dialect.py

```diff
@@ -145,18 +150,33 @@ def create_connect_args(self, url: URL) -> Tuple[List, Dict]:
         """
         Build firebolt-sdk compatible connection arguments.
         URL format : firebolt://id:secret@host:port/db_name
+        For Core: firebolt://db_name?url=http://localhost:8080
+        (full URL including scheme, host, port in url parameter)
         """
         parameters = dict(url.query)
+
+        is_core_connection = "url" in parameters
+        core_url = parameters.pop("url", None) if is_core_connection else None
+
         # parameters are all passed as a string, we need to convert
         # bool flag to boolean for SDK compatibility
         token_cache_flag = bool(strtobool(parameters.pop("use_token_cache", "True")))
-        auth = _determine_auth(url.username, url.password, token_cache_flag)
+
+        if is_core_connection:
+            auth = _determine_auth("", "", token_cache_flag, is_core=True)
+        else:
```

Comment:
This is redundant, pass the whole url to `_determine_auth` so it can extract and decide which auth to use